backend/app/models.py

```diff
@@ -1,6 +1,325 @@
 """
 Models for the ***** app.
 """
-# from django.db import models
+from django.db import models
+import nltk
+import string
+from collections import Counter
+from more_itertools import windowed
 
-# no models yet -- write me!
+
+class Document(models.Model):
+    """
+    This model will hold the full text and
+    metadata (author, title, publication date, etc.) of a document
+    """
+    author = models.CharField(max_length=255, blank=True)
+    date = models.IntegerField(null=True, blank=True)
+    new_attributes = models.JSONField(null=True, blank=True, default=dict)
+    text = models.TextField(blank=True)
+    title = models.CharField(max_length=255, blank=True)
+    word_count = models.PositiveIntegerField(blank=True, null=True, default=None)
+    tokenized_text = models.JSONField(null=True, blank=True, default=None)
+    word_counts_counter = models.JSONField(null=True, blank=True, default=dict)
+    part_of_speech_tags = models.JSONField(null=True, blank=True, default=list)
+
+    def _clean_quotes(self):
+        """
+        Scans through the text and replaces all of the smart quotes and apostrophes with their
+        "normal" ASCII variants
+
+        :param self: The Document to reformat
+        :return: A string that is identical to `text`, except with its smart quotes exchanged
+        """
+
+        # Define the quotes that will be swapped out
+        smart_quotes = {
+            '“': '"',
+            '”': '"',
+            "‘": "'",
+            "’": "'",
+        }
+
+        # Replace all entries one by one
+        output_text = self.text
+        for quote in smart_quotes:
+            output_text = output_text.replace(quote, smart_quotes[quote])
+        self.text = output_text
+        self.save()
+        return self.text
+
+    def get_tokenized_text(self):
+        """
+        Tokenizes the text and returns it as a list of tokens, while removing all punctuation.
+
+        Note: This does not currently properly handle dashes or contractions.
```

Comment:
It looks like there's a few holdovers from the previous version of this method still in place. The docstring note doesn't apply anymore, right? And I assume we can get rid of the conditional guard on line 61, since we'll presumably call this method in some capacity when we `.save()` the model instance.